### PR TITLE
Add more `cudf-polars` unaryops

### PR DESCRIPTION
**Pull request by brandon-b-miller**
This PR adds support for additional unaryops through `cudf-polars`. 

Closes https://github.com/rapidsai/cudf/issues/16566

### CODE REVIEW

File: python/cudf_polars/cudf_polars/dsl/expr.py [hunk 1]
```diff
@@ -932,6 +956,28 @@ def __init__(
...
+            "exp",
+            "log",
+            "sqrt",
+            "cbrt",
+            "ceil",
+            "floor",
+            "abs",
+            "bit_invert",
+            "not",
         ):
```
Comment (wence-) [inline python/cudf_polars/cudf_polars/dsl/expr.py:981]: DRY:
```suggestion
        ) or self.name not in self._MAPPING:
```

File: python/cudf_polars/cudf_polars/dsl/expr.py [hunk 2]
```diff
@@ -1038,6 +1084,24 @@ def do_evaluate(
                 )
                 arg = evaluated.obj_scalar if evaluated.is_scalar else evaluated.obj
             return Column(plc.replace.replace_nulls(column.obj, arg))
+        elif self.name == "pow":
+            column = self.children[0].evaluate(df, context=context, mapping=mapping)
+            if isinstance(self.children[1], Literal):
+                arg = plc.interop.from_arrow(self.children[1].value)
```
Comment (wence-) [inline python/cudf_polars/cudf_polars/dsl/expr.py:1090]: Why special-case the `Literal` case, given that `Literal.evaluate` just does what you have written here for you?

File: python/cudf_polars/cudf_polars/dsl/expr.py [hunk 3]
```diff
@@ -1038,6 +1084,24 @@ def do_evaluate(
...
+                )
+                arg = evaluated.obj_scalar if evaluated.is_scalar else evaluated.obj
+            return Column(
+                plc.binaryop.binary_operation(
+                    column.obj, arg, plc.binaryop.BinaryOperator.POW, self.dtype
+                )
+            )
+        elif self.name in self._MAPPING:
+            column = self.children[0].evaluate(df, context=context, mapping=mapping)
+            casted = plc.unary.cast(column.obj, self.dtype)
```
Comment (wence-) [inline python/cudf_polars/cudf_polars/dsl/expr.py:1103]: *question*: why is this necessary? Are you trying to obtain the correct output type? We certainly don't want to cast if the column is _already_ of the correct dtype.

File: python/cudf_polars/tests/expressions/test_numeric_unaryops.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        "not_",
+    ]
+)
+def op(request):
+    return request.param
+
+
+@pytest.fixture
+def ldf():
+    return pl.DataFrame({"a": [1, 2, None, 4, 5]}).lazy()
```
Comment (wence-) [inline python/cudf_polars/tests/expressions/test_numeric_unaryops.py:42]: Can we test (at least) both float and int inputs?

**Review by wence- — CHANGES_REQUESTED** [review]
There are some xfailed tests that now pass

File: python/cudf_polars/cudf_polars/dsl/expr.py (hunk 3 above)
Comment (brandon-b-miller) [reply python/cudf_polars/cudf_polars/dsl/expr.py:1103]: Yes, this was to obtain the correct datatype. Usually I assume that casting to the same type is a no-op at some lower level, not sure if that's the case for libcudf.

Comment (wence-) [reply python/cudf_polars/cudf_polars/dsl/expr.py:1103]: can you show some examples where this cast is necessary?

Comment (brandon-b-miller) [reply python/cudf_polars/cudf_polars/dsl/expr.py:1103]: For example, `sin` of an integer column in polars gives float while pylibcudf will return the same dtype as what's input. 

```python
>>> import cudf._lib.pylibcudf as plc
>>> import pyarrow as pa
>>> ary = plc.interop.from_arrow(pa.array([1,2,3]))
>>> result = plc.unary.unary_operation(ary, plc.unary.UnaryOperator.SIN)
>>> plc.interop.to_arrow(result)
<pyarrow.lib.ChunkedArray object at 0x7f4112364770>
[
  [
    0,
    0,
    0
  ]
]

>>> result.type().id()
<type_id.INT64: 4>
>>> import polars as pl
>>> df = pl.DataFrame({'a':[1,2,3]}).lazy()
>>> df.select(pl.col('a').sin())
<LazyFrame at 0x7F3D98CF7490>
>>> df.select(pl.col('a').sin()).collect()
shape: (3, 1)
┌──────────┐
│ a        │
│ ---      │
│ f64      │
╞══════════╡
│ 0.841471 │
│ 0.909297 │
│ 0.14112  │
└──────────┘
```

Comment (wence-) [reply python/cudf_polars/cudf_polars/dsl/expr.py]: Thanks.

File: python/cudf_polars/tests/expressions/test_numeric_unaryops.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        "arcsin",
+        "arccos",
+        "arctan",
+        "sinh",
+        "cosh",
+        "tanh",
+        "arcsinh",
+        "arccosh",
+        "arctanh",
+        # "exp", Missing rust side impl
```
Comment (bdice) [inline python/cudf_polars/tests/expressions/test_numeric_unaryops.py:26]: Should this comment reference an issue?

**Review by bdice — APPROVED** [review]
Responding to a review request - this code seems fine from what I know.

File: python/cudf_polars/tests/expressions/test_numeric_unaryops.py (hunk 5 above)
Comment (wence-) [reply python/cudf_polars/tests/expressions/test_numeric_unaryops.py:26]: I did it yesterday: https://github.com/pola-rs/polars/pull/18317

File: python/cudf_polars/cudf_polars/dsl/expr.py [hunk 6]
```diff
@@ -936,7 +959,9 @@ class UnaryFunction(Expr):
             "cum_sum",
         }
     )
-    _supported_fns = frozenset(_supported_misc_fns.union(_supported_cum_aggs))
+    _supported_fns = frozenset(
+        _supported_misc_fns.union(_supported_cum_aggs).union(frozenset(_MAPPING.keys()))
```
Comment (wence-) [inline python/cudf_polars/cudf_polars/dsl/expr.py:963]:
```suggestion
        _supported_misc_fns | _supported_cum_aggs | _MAPPING.keys()
```

(if it works).

If not:
```
frozenset().union(_supported_misc_fns, _supported_cum_aggs, _MAPPING.keys())
```
which definitely does.

File: python/cudf_polars/cudf_polars/dsl/expr.py [hunk 7]
```diff
@@ -1057,6 +1083,23 @@ def do_evaluate(
                 )
                 arg = evaluated.obj_scalar if evaluated.is_scalar else evaluated.obj
             return Column(plc.replace.replace_nulls(column.obj, arg))
+        elif self.name == "pow":
+            column = self.children[0].evaluate(df, context=context, mapping=mapping)
+
+            evaluated = self.children[1].evaluate(df, context=context, mapping=mapping)
```
Comment (wence-) [inline python/cudf_polars/cudf_polars/dsl/expr.py:1089]: `(column, power) = (child.evaluate(df, context=context, mapping=mapping) for child in self.children)`


And then we notice if children is ever too long.

File: python/cudf_polars/tests/expressions/test_numeric_unaryops.py [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+
+@pytest.fixture(params=[pl.Int32, pl.Float32])
+def dtype(request):
+    return request.param
+
+
+@pytest.fixture
+def ldf(dtype):
+    return pl.DataFrame({"a": pl.Series([1, 2, None, 4, 5], dtype=dtype)}).lazy()
```
Comment (wence-) [inline python/cudf_polars/tests/expressions/test_numeric_unaryops.py:46]: Add a `nan` and `inf` as well if it's a float type? Also some negative values, so that we're checking what happens in the `sqrt` case?

File: python/cudf_polars/tests/expressions/test_numeric_unaryops.py [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+
+@pytest.fixture(
+    params=[
+        "sin",
+        "cos",
+        "tan",
+        "arcsin",
+        "arccos",
+        "arctan",
```
Comment (wence-) [inline python/cudf_polars/tests/expressions/test_numeric_unaryops.py:19]: *question*: What happens in these cases for the input values that produce a domain error? Do we just get nans?

**Review by wence- — APPROVED** [review]
tiny niggles, but pretty much ready to go!

**Comment by wence-** [comment]
I pushed some updates and will merge once tests pass

**Comment by wence-** [comment]
Thanks Brandon.
